helm/node-operator-chart/templates/deployment.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: node-operator
+  namespace: giantswarm
+  labels:
+    app: node-operator
+spec:
+  replicas: 1
+  revisionHistoryLimit: 3
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: node-operator
+    spec:
+      volumes:
+      - name: node-operator-configmap
+        configMap:
+          name: node-operator-configmap
+          items:
+          - key: config.yml
+            path: config.yml
+      serviceAccountName: node-operator
+      containers:
+      - name: node-operator
+        image: quay.io/giantswarm/node-operator:[[ .SHA ]]
+        args:
+        - daemon
+        - --config.dirs=/var/run/node-operator/configmap/
+        - --config.files=config
+        volumeMounts:
+        - name: node-operator-configmap
+          mountPath: /var/run/node-operator/configmap/
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8000
+          initialDelaySeconds: 30
+          timeoutSeconds: 1
+        resources:
+          requests:
+            cpu: 100m
```

Comment:
I would make this the same as requests, for that guaranteed foo

Reply:
Why the magic? I just copied this as it is from kvm-operator. Does that mean we should change this everywhere?

Reply:
Let me know when this is important. We can change later. I want to move now.

Reply:
See https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed

If requests == limits, kubernetes can do better with scheduling. If this is the case for kvm-operator too, we can fix it there.

Reply:
I still miss something. What is this doing? What is guaranteed? I couldn't find this info in the docs. What problem does this solve?

Reply:
https://medium.com/google-cloud/quality-of-service-class-qos-in-kubernetes-bb76a89eb2c6 is a good write up

Reply:
Thanks. Will read later.